components/relevancy/src/utils.rs

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use md5::{Digest, Md5};
+use url::{Host, Url};
+
+pub type UrlHash = [u8; 16];
+
+/// Given a URL, extract the part of it that we want to use to identify it.
+///
+/// We currently use the final 3 components of the URL domain.
+///
+/// TODO: decide if this should be 3 or 3 components.
```

Comment:
I think this should follow what components of the URL are used for hash generation from upstream, and that should be passed down from Remote Settings along with other remote assets.

Maybe we can define a mapping (following Github's [terminology](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/about-custom-domains-and-github-pages)) as part of the schema for RS assets:

```rs
enum InputTypes {
    /// `{ host }.{ top domain }`.
    /// Example: `https://www.example.com/` -> `example.com`
    UrlApexDomain,
    /// `{ subdomain }.{ host }.{ top domain }`.
    /// Example: `https://www/example.com` -> `www.example.com`
    UrlFullDomain,
   
    /// Other input types...
}
``` 

WDYT?

Reply:
That makes a lot of sense to me, but probably for future work.  I tried to implement it and realized it wasn't so simple to do.

Reply:
Yup, let's handle this as follow ups.